### PR TITLE
chore: go linter version bump

### PR DESCRIPTION
**Pull request by jvanz**
## Description

To avoid issues with the ireturn checker, this commit updates the go linter version as well as fix all the issues reported by it.

### CODE REVIEW

File: config/crd/bases/policies.kubewarden.io_clusteradmissionpolicies.yaml [hunk 1]
```diff
@@ -108,20 +108,20 @@ spec:
...
-                  and extensions/v1beta1, but \"rules\" only included `apiGroups:[\"apps\"],
-                  apiVersions:[\"v1\"], resources: [\"deployments\"]`, a request to
-                  apps/v1beta1 or extensions/v1beta1 would not be sent to the webhook.
-                  \n - Equivalent: match a request if modifies a resource listed in
-                  rules, even via another API group or version. For example, if deployments
+                  and extensions/v1beta1, but "rules" only included `apiGroups:["apps"],
+                  apiVersions:["v1"], resources: ["deployments"]`, a request to apps/v1beta1
+                  or extensions/v1beta1 would not be sent to the webhook. </li> <li>
+                  Equivalent: match a request if modifies a resource listed in rules,
+                  even via another API group or version. For example, if deployments
```
Comment (fabriziosestito) [inline config/crd/bases/policies.kubewarden.io_clusteradmissionpolicies.yaml:119]: the CRD are being generated again, should we split this in another PR/commit?

Comment (jvanz) [reply config/crd/bases/policies.kubewarden.io_clusteradmissionpolicies.yaml:119]: Ops, I should not add the CRDs in this PR. Sorry.

**Review by fabriziosestito — APPROVED** [review]
